crates/hyperdrive-math/src/short/max.rs

```diff
@@ -318,7 +318,7 @@ impl State {
         let mut maybe_solvency =
             self.solvency_after_short(max_bond_amount, spot_price, checkpoint_exposure);
         if maybe_solvency.is_none() {
-            panic!("Initial guess in `max_short` is insolvent.");
+            panic!("Initial guess in `absolute_max_short` is insolvent.");
```

Comment:
> @[jalextowle](https://github.com/jalextowle) jalextowle [20 hours ago](https://github.com/delvtech/hyperdrive/pull/1005#discussion_r1580134104)
> Didn’t we already check that this isn’t insolvent on line 294? I might be missing something, but I don’t expect this to be solvent
> 
> Member
> Author
> @[slundqui](https://github.com/slundqui) slundqui [19 hours ago](https://github.com/delvtech/hyperdrive/pull/1005#discussion_r1580175165)
> Ah, my mistake, I didn't realize this was in the absolute_max_short function. Is it safe to say here then that if absolute_max_short_guess returns a value that is insolvent, that the pool doesn't allow for a short at all? Or is there a better initial guess we can use here
> 
> Member
> @[jalextowle](https://github.com/jalextowle) jalextowle [14 hours ago](https://github.com/delvtech/hyperdrive/pull/1005#discussion_r1580427013)
> We might be able to use a better initial guess. We could always use 0, or we could allow “cross-overs” by allowing invalid solutions so that we can proceed with iterations. The latter sounds fine to me since Newton’s should get to a solution that is solvent after a few iterations.
> 
> This check was based on my (erroneous) thinking at implementation-time that this algorithm would never cross over the solution

Original conversation from PR, commenting her for visibility

Reply:
Will keep this as is for now, can fix in  follow up PR